hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/test/java/org/apache/hadoop/yarn/server/federation/store/impl/TestSQLFederationStateStore.java

```diff
@@ -76,38 +107,479 @@ public void testSqlConnectionsCreatedCount() throws YarnException {
         FederationStateStoreClientMetrics.getNumConnections());
   }
 
-  @Test(expected = NotImplementedException.class)
   public void testAddReservationHomeSubCluster() throws Exception {
     super.testAddReservationHomeSubCluster();
   }
 
-  @Test(expected = NotImplementedException.class)
   public void testAddReservationHomeSubClusterReservationAlreadyExists() throws Exception {
     super.testAddReservationHomeSubClusterReservationAlreadyExists();
   }
 
-  @Test(expected = NotImplementedException.class)
   public void testAddReservationHomeSubClusterAppAlreadyExistsInTheSameSC() throws Exception {
     super.testAddReservationHomeSubClusterAppAlreadyExistsInTheSameSC();
   }
 
-  @Test(expected = NotImplementedException.class)
   public void testDeleteReservationHomeSubCluster() throws Exception {
     super.testDeleteReservationHomeSubCluster();
   }
 
-  @Test(expected = NotImplementedException.class)
   public void testDeleteReservationHomeSubClusterUnknownApp() throws Exception {
     super.testDeleteReservationHomeSubClusterUnknownApp();
   }
 
-  @Test(expected = NotImplementedException.class)
   public void testUpdateReservationHomeSubCluster() throws Exception {
     super.testUpdateReservationHomeSubCluster();
   }
 
-  @Test(expected = NotImplementedException.class)
   public void testUpdateReservationHomeSubClusterUnknownApp() throws Exception {
     super.testUpdateReservationHomeSubClusterUnknownApp();
   }
+
+  class ReservationHomeSC {
```

Comment:
There is some other Router admin test failing.

Reply:
I will fix it.